src/py/flwr/client/app.py

```diff
@@ -378,6 +388,7 @@ def _on_backoff(retry_state: RetryState) -> None:
 
                     # Retrieve context for this run
                     context = node_state.retrieve_context(run_id=run_id)
+                    context.run_config = get_fused_config(run_info[run_id], flwr_dir)
```

Comment:
We should not re-create this each time we're receiving a message. It should be sufficient to do it once for each run.